pkg/controller/vitessshardreplication/vitessshardreplication_controller.go

```diff
@@ -193,22 +193,10 @@ func (r *ReconcileVitessShard) Reconcile(request reconcile.Request) (reconcile.R
 	// multi-step Vitess cluster management workflows.
 	wr := wrangler.New(logutil.NewConsoleLogger(), ts.Server, tmc)
 
-	// Check if we need to initialize the shard.
-	// If it's already initialized, this will be a no-op.
-	// If we are using external MySQL we will bail out early.
-	ismResult, err := r.initShardMaster(ctx, vts, wr)
-	resultBuilder.Merge(ismResult, err)
-
-	// Check if we need to externally reparent
-	// in the case of external MySQL.
-	// If we are not using external MySQL we will bail out early.
-	terResult, err := r.tabletExternallyReparent(ctx, vts, wr)
-	resultBuilder.Merge(terResult, err)
-
-	// Check if we need to start replication on a shard that's been restored
-	// from backup. If it's already initialized, this will be a no-op.
-	irsResult, err := r.initRestoredShard(ctx, vts, wr)
-	resultBuilder.Merge(irsResult, err)
+	// If we specified the operator to handle the initialization of replication, make the appropriate checks.
+	if *vts.Spec.InitReplication {
+		r.initReplication(ctx, vts, wr, resultBuilder)
```

Comment:
See my comments below. Once you are returning a `(reconcile.Result, error)`, you should merge here and take away the conditional entirely:
```suggestion
		initReplicationResult, err := r.initReplication(ctx, vts, wr, resultBuilder)
		resultBuilder.Merge(initReplicationResult, err)
```

Reply:
Done.